pallets/moonbeam-orbiters/src/lib.rs

```diff
@@ -0,0 +1,485 @@
+//! collator in turn with the other orbiters (rotation every `RotatePeriod` rounds).
+//!
+//! This pallet is designed to work with the nimbus consensus.
+//! In order not to impact the other pallets (notably nimbus and parachain-staking) this pallet
+//! simply redefines the lookup NimbusId-> AccountId, in order to replace the collator by its
+//! currently selected orbiter.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod migrations;
+pub mod types;
+
+#[cfg(any(test, feature = "runtime-benchmarks"))]
+mod benchmarks;
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub use pallet::*;
+pub use types::*;
+
+use frame_support::pallet;
+use nimbus_primitives::{AccountLookup, NimbusId};
+
+#[pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_support::traits::{Currency, Imbalance, NamedReservableCurrency};
+	use frame_system::pallet_prelude::*;
+	use sp_runtime::traits::{One, Saturating, StaticLookup, Zero};
+
+	#[pallet::pallet]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	pub type BalanceOf<T> =
+		<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+	pub type ReserveIdentifierOf<T> = <<T as Config>::Currency as NamedReservableCurrency<
+		<T as frame_system::Config>::AccountId,
+	>>::ReserveIdentifier;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// A type to convert between AuthorId and AccountId. This pallet wrap the lookup to allow
+		/// orbiters authoring.
+		type AccountLookup: AccountLookup<Self::AccountId>;
+
+		/// Origin that is allowed to add a collator in orbiters program.
+		type AddCollatorOrigin: EnsureOrigin<Self::Origin>;
+
+		/// The currency type.
+		type Currency: NamedReservableCurrency<Self::AccountId>;
+
+		/// Origin that is allowed to remove a collator from orbiters program.
+		type DelCollatorOrigin: EnsureOrigin<Self::Origin>;
+
+		/// Maximum number of orbiters per collator.
+		type MaxPoolSize: Get<u32>;
+
+		/// Maximum number of round to keep on storage.
+		type MaxRoundArchive: Get<Self::RoundIndex>;
+
+		/// Reserve identifier for this pallet instance.
+		type OrbiterReserveIdentifier: Get<ReserveIdentifierOf<Self>>;
+
+		/// Number of rounds before changing the selected orbiter.
+		type RotatePeriod: Get<Self::RoundIndex>;
+
+		/// Round index type.
+		type RoundIndex: Parameter
+			+ Member
+			+ MaybeSerializeDeserialize
+			+ sp_std::fmt::Debug
+			+ Default
+			+ sp_runtime::traits::MaybeDisplay
+			+ sp_runtime::traits::AtLeast32Bit
+			+ Copy;
+
+		/// Origin that is allowed to update the minimal orbiter deposit amount.
+		type UpdateMinOrbiterDepositOrigin: EnsureOrigin<Self::Origin>;
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn account_lookup_override)]
+	/// Account lookup override
+	pub type AccountLookupOverride<T: Config> =
```

Comment:
I would remove the override from the name. It is the account lookup of Orbiter (yes it overrides the pallet staking, but this type belongs to orbiter pallet)

Reply:
The config takes as associated type an AccountLookup, and adds an overlay to it, no matter where the original lookup comes from (which in our case comes from author-mapping and not from staking), anyway we can't take the same name because it is an overlay.